wp-content/civi-extensions/goonjcustom/Civi/InductionService.php

```diff
@@ -447,4 +447,133 @@ public static function createInductionForTransitionedVolunteer(string $op, strin
+          ->addWhere('activity_type_id:name', '=', 'Email')
+          ->addWhere('subject', '=', $template['msg_subject'])
+          ->addWhere('source_contact_id', '=', $activity['source_contact_id'])
+          ->execute()->single();
+  
+        if (!$emailActivities) {
+          $emailParams = [
+            'contact_id' => $activity['source_contact_id'],
+            'template_id' => $template['id'],
+          ];
+          civicrm_api3('Email', 'send', $emailParams);
+        }
+      }
+  
+      // Move to the next batch by increasing the offset
+      $offset += $batchSize;
+  
+    } while (count($unscheduledInductionActivities) === $batchSize);
+  }
+
+  /**
+   *
+   */
+  public static function updateInductionStatusNoShow() {
+    \Civi::log()->info('cjecka');
+		$followUpDays = 30;
+		$followUpTimestamp = strtotime("-$followUpDays days");
+		$batchSize = 25;
+		$offset = 0;
+
+		try {
+			// Fetch the follow-up message template
+			$template = MessageTemplate::get(FALSE)
+				->addSelect('id', 'msg_subject')
+				->addWhere('msg_title', 'LIKE', 'Induction_slot_booking_follow_up_email%')
+				->execute()->single();
+
+			if (!$template) {
+				throw new \Exception('Follow-up email template not found.');
+			}
+
+			do {
+				// Fetch email activities older than 30 days
+				$followUpEmailActivities = Activity::get(TRUE)
+					->addSelect('source_contact_id', 'activity_date_time')
+					->addWhere('subject', '=', $template['msg_subject'])
+					->addWhere('activity_type_id:label', '=', 'Email')
+					->addWhere('created_date', '<', date('Y-m-d H:i:s', $followUpTimestamp))
+					->setLimit($batchSize)
+					->setOffset($offset)->execute();
+
+				foreach ($followUpEmailActivities as $activity) {
+					// Fetch the associated induction activity
+					$inductionActivity = Activity::get(FALSE)
+						->addSelect('id', 'source_contact_id', 'status_id:name')
+						->addWhere('activity_type_id:name', '=', 'Induction')
+						->addWhere('source_contact_id', '=', $activity['source_contact_id'])
+						->execute()
+						->single();
+
+					if (!$inductionActivity) {
+						\Civi::log()->info('No induction activity found for source contact', [
+							'source_contact_id' => $activity['source_contact_id'],
+						]);
+						continue;
+					}
+
+					// Check if the status is 'Completed' or 'Scheduled'
+					if (in_array($inductionActivity['status_id:name'], ['Completed', 'Scheduled'])) {
+						\Civi::log()->info('Induction activity is already completed or scheduled, skipping', [
+							'inductionActivityId' => $inductionActivity['id'],
+							'status' => $inductionActivity['status_id:name']
+						]);
+						continue;
+					}
+
+					// Update the induction status to 'No_show'
+					$updateResult = Activity::update(FALSE)
+						->addValue('status_id:name', 'No_show')
+						->addWhere('id', '=', $inductionActivity['id'])
+						->execute();
+
+					\Civi::log()->info('Induction activity status updated to No_show', [
```

Comment:
If you are using the try-catch, why are you adding logs everywhere? Let's use the catch block to fetch the necessary logs you want in case the code won't run. 
Whenever this cron runs every time these logs print and I expect we don't need that we only need the logs when the code won't run right?